test/functional/launch/nested_direct/tests/test-launch-nested-Direct.hpp

```diff
@@ -0,0 +1,198 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-23, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef __TEST_LAUNCH_NESTED_DIRECT_HPP__
+#define __TEST_LAUNCH_NESTED_DIRECT_HPP__
+
+#include <numeric>
+
+template <typename INDEX_TYPE, typename WORKING_RES, typename LAUNCH_POLICY,
+          typename THREAD_X_POLICY, typename THREAD_Y_POLICY, typename THREAD_Z_POLICY,
+          typename TEAM_X_POLICY, typename TEAM_Y_POLICY, typename TEAM_Z_POLICY>
+void LaunchNestedDirectTestImpl(INDEX_TYPE M)
+{
+
+  RAJA::TypedRangeSegment<INDEX_TYPE> r1(0, 2*RAJA::stripIndexType(M));
+  RAJA::TypedRangeSegment<INDEX_TYPE> r2(0, 3*RAJA::stripIndexType(M));
+  RAJA::TypedRangeSegment<INDEX_TYPE> r3(0, 4*RAJA::stripIndexType(M));
+
+  RAJA::TypedRangeSegment<INDEX_TYPE> r4(0, 4*RAJA::stripIndexType(M));
+  RAJA::TypedRangeSegment<INDEX_TYPE> r5(0, 5*RAJA::stripIndexType(M));
+  RAJA::TypedRangeSegment<INDEX_TYPE> r6(0, 6*RAJA::stripIndexType(M));
+
+  INDEX_TYPE N1 = static_cast<INDEX_TYPE>(r1.end() - r1.begin());
+  INDEX_TYPE N2 = static_cast<INDEX_TYPE>(r2.end() - r2.begin());
+  INDEX_TYPE N3 = static_cast<INDEX_TYPE>(r3.end() - r3.begin());
+
+  INDEX_TYPE N4 = static_cast<INDEX_TYPE>(r4.end() - r4.begin());
+  INDEX_TYPE N5 = static_cast<INDEX_TYPE>(r5.end() - r5.begin());
+  INDEX_TYPE N6 = static_cast<INDEX_TYPE>(r6.end() - r6.begin());
+
+
+  INDEX_TYPE N = static_cast<INDEX_TYPE>(RAJA::stripIndexType(N1) *
```

Comment:
Is it necessary to strip these index values since they are already defined as INDEX_TYPE? Is this because we don't support multiplication for type-safety types?

Reply:
Actually this does work, not sure why I thought it did not....